docs/rules/0121/no-cycles.md

```diff
@@ -0,0 +1,129 @@
+---
+rule:
+  aip: 121
+  name: [core, '0121', no-cycles]
+  summary: Resources must not form a resource reference cycle.
+permalink: /121/no-cycles
```

Comment:
how about naming the rule `no-mutable-cycles`? it might be clearer. 

At least in conversations with folks there's always a question on whether output-only cycles are allowed.

Reply:
For sure, I can do this, just after all other comments are resolved as I have to rename everything.

Reply:
Done.